libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2311,16 +2311,16 @@ string YulUtilFunctions::copyArrayFromStorageToMemoryFunction(ArrayType const& _
 				function <functionName>(slot) -> memptr {
 					memptr := <allocateTemp>()
 					let end := <encode>(slot, memptr)
-					mstore(<freeMemoryPointer>, end)
+					<finalize>(memptr, sub(end, memptr))
 				}
 			)")
 			("functionName", functionName)
-			("allocateTemp", allocationTemporaryMemoryFunction())
+			("allocateTemp", allocateUnboundedFunction())
 			(
 				"encode",
 				abi.abiEncodeAndReturnUpdatedPosFunction(_from, _to, ABIFunctions::EncodingOptions{})
 			)
-			("freeMemoryPointer", to_string(CompilerUtils::freeMemoryPointer))
+			("finalize", finalizeAllocationFunction())
```

Comment:
I think `finalizeAllocation` would be better throughout. With `finalize` one needs to look up what it means.